src/Bindings/ManualBindings.cpp

```diff
@@ -1002,7 +1052,14 @@ static int tolua_cPluginManager_AddHook_FnRef(cPluginManager * a_PluginManager,
 	}
 
 	// Add the hook to the plugin
-	if (!Plugin->AddHookRef(HookType, a_ParamIdx + 1))
+	auto callback = std::make_shared<cLuaState::cCallback>();
```

Comment:
Why does this need to be a shared pointer? Its only owned by one thread then copied into the hookmap. We could just pass my reference into GetStackValue + AddHookCallback, then move into the hookmap.